src/dune_lang/package_dependency.ml

```diff
@@ -40,12 +40,64 @@ let encode { name; constraint_ } =
   | Some c -> pair Package_name.encode Package_constraint.encode (name, c)
 ;;
 
+(* Check for common typos in package dependency constraints *)
+let check_for_typo ~loc { name; constraint_ } =
+  (* Only check for typos if warnings are enabled *)
+  if Dune_config.Config.(get typo_warnings) = `Enabled
```

Comment:
I would probably flip the condition here so the `None` is the early out option as then you can then mentally concentrate on the other option, without the `None` option hanging out at the very end of the code.